scripts/scil_generate_sampling_scheme.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import division
+
+import argparse
+import logging
+import numpy as np
+
+from scilpy.samplingscheme.gen_scheme import gen_scheme
+from scilpy.samplingscheme.optimize_scheme import (add_b0s,
+                                                   add_bvalue_b0,
+                                                   bvalue_lin_b,
+                                                   bvalue_lin_q,
+                                                   correct_b0s_philips,
+                                                   min_duty_cycle_bruteforce,
+                                                   swap_sampling_eddy)
+from scilpy.samplingscheme.save_scheme import (save_scheme_bvecs_bvals,
+                                               save_scheme_caru,
+                                               save_scheme_philips,
+                                               save_scheme_siemens,
+                                               save_scheme_mrtrix)
+
+
+DESCRIPTION = """
+Generate multi-shell sampling schemes with various processing to accelerate
+acquisition and help artefact correction.
+
+Multi-shell schemes are generated as in [1], the bvecs are then flipped
+to maximize spread for eddy current correction, b0s are interleaved
+at equal spacing and the non-b0 samples are finally shuffled
+to minimize the total diffusion gradient amplitude over a few TR.
+    """
+
+EPILOG = """
+References: [1] Emmanuel Caruyer, Christophe Lenglet, Guillermo Sapiro,
+Rachid Deriche. Design of multishell sampling schemes with uniform coverage
+in diffusion MRI. Magnetic Resonance in Medicine, Wiley, 2013, 69 (6),
+pp. 1534-1540. <http://dx.doi.org/10.1002/mrm.24736>
+    """
+
+
+def buildArgsParser():
+    p = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=DESCRIPTION,
+        epilog=EPILOG)
+    p._optionals.title = "Options and Parameters"
+
+    p.add_argument(
+        'ks', metavar='Ks', type=int, nargs='+',
+        help='Number of samples on each shells.')
+    p.add_argument(
+        'outfile', action='store', metavar='outfile', type=str,
```

Comment:
Please have a go through all the arguments to remove the unneeded args (based on the default values, `type`, `action` and `metavar`)